packages/ai/core/prompt/convert-to-language-model-prompt.ts

```diff
@@ -388,3 +200,131 @@ async function downloadAssets(
     downloadedImages.map(({ url, data }) => [url.toString(), data]),
   );
 }
+
```

Comment:
Rewriting this was not strictly necessary, but given the complexity of what was there before, this made things much easier to understand.

A future enhancement here might be to normalize the `image` and `file`. But it's probably better to wait to do this until there are more examples of file and image uploading in other SDKs.